src-tauri/src/gui/commands/settings.rs

```diff
@@ -220,20 +222,56 @@ pub fn get_idf_mirror_list(app_handle: AppHandle) -> Value {
   };
 
   let mirror = settings.idf_mirror.clone().unwrap_or_default();
-  let mut available_mirrors = idf_im_lib::get_idf_mirrors_list().to_vec();
+  let mut available_mirrors: Vec<String> = idf_im_lib::get_idf_mirrors_list()
```

Comment:
this is unnecessary allocation, you can work with the &str, no need to copy it over.